### PR TITLE
Remove Calico felix override workaround

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**: Since Calico v3.25 is out, we should no longer need the workaround for https://github.com/projectcalico/calico/issues/3145 as the fix is now released. This PR removes it.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: Tiltfile [hunk 1]
```diff
@@ -363,7 +363,7 @@ def deploy_worker_templates(template, substitutions):
         calico_values = "./templates/addons/calico-dual-stack/values.yaml"
     else:
         calico_values = "./templates/addons/calico/values.yaml"
-    flavor_cmd += "; " + helm_cmd + " repo add projectcalico https://projectcalico.docs.tigera.io/charts; " + helm_cmd + " --kubeconfig ./${CLUSTER_NAME}.kubeconfig install calico projectcalico/tigera-operator -f " + calico_values + " --namespace tigera-operator --create-namespace; kubectl --kubeconfig ./${CLUSTER_NAME}.kubeconfig apply -f ./templates/addons/calico/felix-override.yaml"
+    flavor_cmd += "; " + helm_cmd + " repo add projectcalico https://projectcalico.docs.tigera.io/charts; " + helm_cmd + " --kubeconfig ./${CLUSTER_NAME}.kubeconfig install calico projectcalico/tigera-operator -f " + calico_values + " --namespace tigera-operator --create-namespace"
```
Comment (CecileRobertMichon) [inline Tiltfile:366]: /hold

to validate Tilt

Comment (CecileRobertMichon) [reply Tiltfile:366]: Tilt looks good

/hold cancel

**Comment by jackfrancis** [comment]
/lgtm
/approve

🧹

**Pull request merged**